gui/default/syncthing/core/syncthingController.js

```diff
@@ -1686,17 +1744,234 @@ angular.module('syncthing.core')
+                            resp.data.publicKey.excludeCredentials = (($scope.webauthn.state || {}).credentials || []).map(function (cred) {
+                                return { type: "public-key", id: cred.id };
+                            });
+                            return webauthnJSON.create(resp.data);
+                        })
+                        .then(function (pkc) {
+                            return $http.post(urlbase + '/webauthn/register-finish', pkc);
+                        })
+                        .then(function (resp) {
+                            if (!$scope.webauthn.state) {
+                                $scope.webauthn.state = {};
+                            }
+                            if (!$scope.webauthn.state.credentials) {
+                                $scope.webauthn.state.credentials = [];
+                            }
+                            $scope.webauthn.state.credentials.push(resp.data);
+                        })
+                        .catch(function (e) {
+                            if (e instanceof DOMException && e.name === "InvalidStateError") {
+                                $scope.webauthn.errors.alreadyRegistered = true;
+                            } else if (e instanceof DOMException && e.name === "AbortError") {
+                                $scope.webauthn.errors.aborted = true;
+                            } else if (e instanceof DOMException && e.name === "NotAllowedError") {
+                                $scope.webauthn.errors.notAllowed = true;
+                            } else if (e instanceof DOMException && e.name === "SecurityError") {
+                                $scope.webauthn.errors.securityError = true;
+                            } else {
+                                $scope.webauthn.errors.registrationFailed = true;
+                                console.log('Credential creation failed:', e);
+                            }
+                        });
+                };
+
+                $scope.deleteWebauthnCredential = function (cred) {
+                    $scope.webauthn.state.credentials = $scope.webauthn.state.credentials.filter(function (cr) {
+                        return cr.id !== cred.id;
+                    });
+                };
+
+                $scope.saveWebauthnState = function () {
+                    return $http.post(
+                        urlbase + '/webauthn/state',
+                        $scope.webauthn.state,
+                        { headers: { 'Content-Type': 'application/json' } }
+                    ).then(function() {
+                        $scope.webauthn.editingCredentialIds = {};
+                    }).finally(function () {
+                        console.log('saveWebauthnState', $scope.webauthn.state);
+                        refreshWebauthnState();
+                    });
+                };
+
+            } else {
+                // Functions for use on the login page
```

Comment:
Is it really necessary to filter out the unused functions from the `$scope`?  Do they hurt?  I mean, we have tons of functions irrelevant to the login page, but still always define them.

Reply:
Probably not. Fixed in b433ef668a871bad8542b79ab6df50255007f62b.